Editor/Analytics/AnalyticsUtils.cs

```diff
@@ -20,32 +20,54 @@ internal static AnalyticsState GetTrackingState()
             {
                 return AnalyticsState.Disabled;
             }
-            return EditorPrefExtensions.GetEnum(KeyTrackingState, AnalyticsState.Unknown);
+
+            try
+            {
+                return EditorPrefExtensions.GetEnum(KeyTrackingState, AnalyticsState.Unknown);
+            }
+            catch (ArgumentException)
+            {
+                EditorPrefExtensions.SetEnum(KeyTrackingState, AnalyticsState.Unknown);
+                return AnalyticsState.Unknown;
+            }
         }
 
         internal static void SetTrackingTo(AnalyticsState state)
         {
-            if (state == AnalyticsState.Minimal)
+            AnalyticsState currentState = GetTrackingState();
+            switch (state)
```

Comment:
Please leave this if statements, this does look confusing as hell, only after the 4th or 5th readthrough i understood the cases are also if cases... :x

Reply:
Interesting. For me the switch statement is much more readable then the if else clauses. But I will change it :D